git.go

```diff
@@ -81,15 +90,18 @@ func (g *GitClient) Pull(uri, branch string, depth int) error {
 	if depth > 0 {
 		args = append(args, "--depth", strconv.Itoa(depth))
 	}
+	args = append(args, "--recurse-submodules")
 	cmd := g.command("git", args...)
 
 	// Discard output to have zero chance of logging the access token.
 	cmd.Stdout = ioutil.Discard
 	cmd.Stderr = ioutil.Discard
 
 	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("clone failed: %s", err)
+		return fmt.Errorf("pull failed: %s", err)
 	}
+	submodulesGet := g.command("git", "submodule", "update", "--init", "--recursive")
```

Comment:
This would be a breaking change for existing users I believe. As such I think it needs to default to _not_ being run, and then use a flag to toggle it on?

Reply:
I've used this an all my jobs involving PR, and git just ignores all excessive parameters. 
You can try out my build [djirik/github-pr-resource](https://hub.docker.com/r/djirik/github-pr-resource). I'll see what I can to about flag